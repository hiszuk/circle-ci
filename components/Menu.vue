<template>
  <div class="menu">
    <MenuItem class="menu-item"
      v-for="item in items"
      :key = "item.label"
      v-bind = "item"
      @clickMenuItem = "onClickMenuItem"
    />
  </div>
</template>

<script>
import MenuItem from '~/components/MenuItem'
export default {
  name: 'Menu',
  components: {
    MenuItem
  },
  props: {
    items: {
      type: Array,
      default: [],
      required: true
    }
  },
  methods: {
    onClickMenuItem({ name }) {
      this.$emit('clickMenuItem', { name })
    },
  },
}
</script>