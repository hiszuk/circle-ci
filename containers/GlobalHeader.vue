<template>
  <header class="global-header">
    <span class="global-header-title" @click="navigateRoot">
      <SiteTitle :title="siteTitle" />
    </span>
    <Menu
      class="global-header-menu"
      :items="menuItems"
      @clickMenuItem="onClickMenuItem"
    />
  </header>
</template>

<script>
import { mapState } from "vuex";
import SiteTitle from "~/basics/SiteTitle";
import Menu from "~/components/Menu";
export default {
  name: 'GlobalHeader',
  components: {
    SiteTitle,
    Menu,
  },
  computed: {
    ...mapState(["siteTitle", "menuItems"]),
  },
  methods: {
    onClickMenuItem({ name }) {
      this.$emit('navigate', { name })
    },
    navigateRoot() {
      this.$emit('navigate', { name: '/'})
    },
  },
}
</script>
