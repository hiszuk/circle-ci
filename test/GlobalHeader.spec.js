import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import GlobalHeader from "@/containers/GlobalHeader.vue";
import Menu from "@/components/Menu.vue";
import menuItems from "@/test/_mockData/menuItems.json";

const localVue = createLocalVue();
localVue.use(Vuex);
const factory = ({ store, localVue }) => {
  return shallowMount(GlobalHeader, {
    store,
    localVue
  });
};

describe("GlobalHeader.vue", () => {
  // 初期化
  let store;
  beforeAll(() => {
    store = new Vuex.Store({
      state: {
        siteTitle: "test site title",
        menuItems: menuItems
      }
    });
  });

  describe("methods確認", () => {
    it("click menu item", () => {
      const wrapper = factory({ store, localVue });
      // method起動
      wrapper.vm.onClickMenuItem(menuItems[0]);
      expect(wrapper.emitted("navigate")).toBeTruthy();
      expect(wrapper.emitted("navigate")[0][0]).toEqual({
        name: menuItems[0].name
      });

      wrapper.vm.onClickMenuItem(menuItems[1]);
      expect(wrapper.emitted("navigate")).toBeTruthy();
      expect(wrapper.emitted("navigate")[1][0]).toEqual({
        name: menuItems[1].name
      });
    });

    it("click root", () => {
      const wrapper = factory({ store, localVue });
      wrapper.vm.navigateRoot();
      expect(wrapper.emitted("navigate")).toBeTruthy();
      expect(wrapper.emitted("navigate")[0][0]).toEqual({
        name: "/"
      });
    });
  });

  describe("template確認", () => {
    it("サイトタイトルクリック時にmethodがコールされるか", () => {
      const wrapper = factory({ store, localVue });
      wrapper.find(".global-header-title").trigger("click")
      expect(wrapper.emitted("navigate")).toBeTruthy();
      expect(wrapper.emitted("navigate")[0][0]).toEqual({
        name: "/"
      });
    })

    it("snapshot", () => {
      const wrapper = factory({ store, localVue });
      // レンダリング結果がスナップショットと同じか
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });
});
