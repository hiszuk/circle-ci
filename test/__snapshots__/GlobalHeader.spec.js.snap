// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalHeader.vue template確認 snapshot 1`] = `
<header
  class="global-header"
>
  <span
    class="global-header-title"
  >
    <sitetitle-stub
      title="test site title"
    />
  </span>
   
  <menu-stub
    class="global-header-menu"
    items="[object Object],[object Object]"
  />
</header>
`;
